src/Controller/AdminRegistrationController.php

```diff
@@ -77,7 +105,65 @@ public function finishRegistration(HTTPRequest $request): HTTPResponse
 
         $store::clear($request);
 
-        return $this->jsonResponse(['success' => true], 201);
+        return $this->jsonResponse([
+            'success' => true,
+            'method' => $result->getContext()['registeredMethod'] ?? null,
+        ], 201);
+    }
+
+    /**
+     * Remove the specified method from the currently logged in user
+     *
+     * @param HTTPRequest $request
+     * @return HTTPResponse
+     */
+    public function removeRegisteredMethod(HTTPRequest $request): HTTPResponse
+    {
+        // Ensure CSRF protection
+        if (!SecurityToken::inst()->checkRequest($request)) {
+            return $this->jsonResponse(
+                ['errors' => [_t(__CLASS__ . '.CSRF_FAILURE', 'Request timed out, please try again')]],
+                400
+            );
+        }
+
+        // Get the specified method
+        $methodRegistry = MethodRegistry::singleton();
+        $method = $request->param('Method');
+
+        if (!$method || !($method = $methodRegistry->getMethodByURLSegment($request->param('Method')))) {
```

Comment:
Presumably the second half of this condition would return false with a null argument anyway, so we need the first bit?

Reply:
Strict typing requires that we don't pass in `null` - so we need the check. I can probably make this a little nicer by not re-declaring the same named variable though so I'll do that.